src/test/client/TestClient.h

```diff
@@ -20,8 +20,70 @@
 #include "msg/Messenger.h"
 #include "mon/MonClient.h"
 #include "osdc/ObjectCacher.h"
-
+#include "client/MetaRequest.h"
 #include "client/Client.h"
+#include "messages/MClientReclaim.h"
+#include "messages/MClientSession.h"
+#include "common/async/blocked_completion.h"
+
+#define dout_subsys ceph_subsys_client
+
+namespace bs = boost::system;
+namespace ca = ceph::async;
+
+class ClientScaffold : public Client {  
+public:
+    ClientScaffold(Messenger *m, MonClient *mc, Objecter *objecter_) : Client(m, mc, objecter_) {}
+    ~ClientScaffold();
+    int check_dummy_op(const UserPerm& perms){
+      RWRef_t mref_reader(mount_state, CLIENT_MOUNTING);
+      if (!mref_reader.is_state_satisfied()) {
+        return -CEPHFS_ENOTCONN;
+      }
+      MetaRequest *req = new MetaRequest(CEPH_MDS_OP_DUMMY);
+      int res = make_request(req, perms);
+      ldout(cct, 10) << __func__ << " result=" << res << dendl;
+      return res;
+    }
+    int check_unknown_session_op() {
+      RWRef_t mref_reader(mount_state, CLIENT_MOUNTING);
+      if (!mref_reader.is_state_satisfied()) {
+        return -CEPHFS_ENOTCONN;
+      }
+      auto session = _get_or_open_mds_session(0);
+      auto msg = make_message<MClientSession>(-1, session->seq);
+      int res = session->con->send_message2(std::move(msg));
+      ldout(cct, 10) << __func__ << " result=" << res << dendl;
+      return res;
+    }
+    bool check_client_blocklisted() {
+      objecter->maybe_request_map();
+      bool is_blocklisted = false;
+      for(int i = 0; i < 5; ++i) {
+        const auto myaddrs = messenger->get_myaddrs();
+        is_blocklisted = objecter->with_osdmap([&](const OSDMap& o) {
+                                   return o.is_blocklisted(myaddrs);});
+        if(is_blocklisted) {
+          break;
+        }                           
+        sleep(1);                           
+      }
```

Comment:
This `is_blocklisted` logic is already in `Client::handle_osd_map`. You just need to add a protected method in `Client` to access the `Client::blocklisted` member.

Reply:
Right, I added a method `Client::is_client_blocklisted` that would return `Client:blocklisted`. Thanks.